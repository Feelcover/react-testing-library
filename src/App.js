import { useState, useEffect } from "react";
import "./App.css";

function App() {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);
  const onClick = (e) => {
    setToggle(!toggle);
  };
  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 100);
  }, []);
  return (
    <div className="App">
      {toggle && <div data-testid="toggle-elem">TOGGLE</div>}
      {data && <div>qweqwe</div>}
      <h1>Hello world</h1>
      <button data-testid="toggle-btn" onClick={onClick}>click</button>
      <input type="text" placeholder="введите значение..." />
    </div>
  );
}

export default App;
