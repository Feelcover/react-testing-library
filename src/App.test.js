import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";

describe("app test", () => {
  test("renders learn react link", async () => {
    render(<App />);
    // const helloWorldEl = screen.getByText(/hello world/i);
    // const button = screen.getByRole("button");
    // const input = screen.getByPlaceholderText(/введите значение.../i);
    // expect(helloWorldEl).toBeInTheDocument();
    // expect(button).toBeInTheDocument();
    // expect(input).toMatchSnapshot();
    screen.debug();
     const helloWorldEl = await screen.findByText(/qweqwe/i);
     expect(helloWorldEl).toBeInTheDocument();
    screen.debug();
  }),
  test("click", async () => {
    render(<App />);
    const btn = screen.getByTestId("toggle-btn");
    expect(screen.queryByTestId("toggle-elem")).toBeNull();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle-elem")).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle-elem")).toBeNull();
  }),
  test("input", () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/введите значение.../i);
    screen.debug();
    expect(screen.queryByTestId("value-elem")).toContainHTML("");
    screen.debug();
    fireEvent.input(input, {
      target: { value: "qweqwe" },
    });
    expect(screen.queryByTestId("value-elem")).toContainHTML('qweqwe');
  });
});
