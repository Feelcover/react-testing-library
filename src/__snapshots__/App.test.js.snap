// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders learn react link 1`] = `
<input
  placeholder="введите значение..."
  type="text"
/>
`;

exports[`app test renders learn react link 1`] = `
<input
  placeholder="введите значение..."
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="введите значение..."
  type="text"
/>
`;
